runtime/interpreter/variable.go

```diff
@@ -36,13 +38,19 @@ func (v *Variable) SetValue(value Value) {
 	v.value = value
 }
 
-func NewVariableWithValue(value Value) *Variable {
+func NewVariableWithValue(gauge common.MemoryGauge, value Value) *Variable {
+	if gauge != nil {
```

Comment:
`common.UseMemory()` method already have this `nil` check. should be safe to pass `gauge` as is